plugins/woocommerce-blocks/docs/third-party-developers/extensibility/checkout-block/additional-checkout-fields.md

```diff
@@ -0,0 +1,531 @@
+
+There are plans to expand this list, but for now these are the types available.
+
+## Using the API
+
+To register additional checkout fields you must use the `woocommerce_blocks_register_checkout_field` function.
+
+It is recommended to run this function after the `woocommerce_loaded` action.
+
+The registration function takes an array of options describing your field. Some field types take additional options.
+
+### Options
+
+#### General options
+
+These options apply to all field types (except in a few circumstances which are noted inline).
+
+| Option name     | Description                                                                                                                            | Required? | Example                                      | Default value                                                              |
+|---------------|----------------------------------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------|-----------|----------------------------------------------------------------------------|
+| `id`            | The field's ID. This should be a unique identifier for your field. It is composed of a namespace and field name separated by a `/`.    | Yes       | `plugin-namespace/how-did-you-hear`          | No default - this must be provided.                                        |
+| `label`         | The label shown on your field. This will be the placeholder too.                                                                       | Yes       | `How did you hear about us?`                 | No default - this must be provided.                                        |
+| `optionalLabel` | The label shown on your field if it is optional. This will be the placeholder too.                                                     | No        | `How did you hear about us? (Optional)`      | The default value will be the value of `label` with `(optional)` appended. |
+| `location`      | The location to render your field.                                                                                                     | Yes       | `contact`, `address`, or `additional`        | No default - this must be provided.                                        |
+| `type`          | The type of field you're rendering. It defaults to `text` and must match one of the supported field types.                             | No        | `text`, `select`, or `checkbox`              | `text`                                                                     |
+| `attributes`    | An array of additional attributes to render on the field's input element. This is _not_ supported for `select` fields.                 | No        | `[	'data-custom-data' => 'my-custom-data' ]` | `[]`                                                                       |
```

Comment:
`sanitize_callback` A function called to sanitize the customer provided value when posted.
`validate_callback` A functionc a called to validate the customer provided value when posted, after sanitization.